WordLadder.java

```diff
@@ -0,0 +1,47 @@
+import java.util.*;
+
+public class WordLadder {
+    // BFS Approach
+    /*
+        For each character in beginWord, try changing the letter
+        if it is there in the set, it is one of the words in the dictionary
+        remove it from the set and add it to the queue
+        Each word in the queue will have the same process and if we find a match with endWord, we got the answer
+        Why? step + 1 -> we are expected to return the number of words in the transformation
+        e.g. hit -> hot   Here, we got the answer in two steps but in the code, we are incrementing that value only if we pick a new element from the queue
+     */
+    // TC : O(l * w) -> l - length of each word  w - number of words
```

Comment:
Time complexity will be k*n^2 --- n is #words, O(n) for stringCombinations and k is avg size of word